examples/apical_point.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+'''example for finding the apical point, and potentially plotting it'''
+import argparse
+import logging
+
+import neurom as nm
+from neurom.core.dataformat import COLS
+from neurom.morphmath import point_dist2
+
+L = logging.getLogger(__name__)
+
+
+def get_apical_point(morph, tuft_percent=20):
+    '''Attempt to find the apical point in 'tufted' neurons
+
+    Consider a neuron:
+
+        |   /    | Tuft = 20%
+        |--/     |
+        |   /
+        |--/
+        |
+    ----.-----
+
+    All endpoints in the top 'tuft_percent' are found, then their common
+    branch segment, furthest from the soma, is identified.
+
+    Args:
+        morph: neurom.fst._core.Neuron
+        tuft_percent: percentage of the 'height' of the apical dendrite that
+        would enclose the tuft, only leaves in this volume are considered as
+        endpoints.  Note that this a spherical shell centered at the soma
+
+    Returns:
+        Section whose *end* is where the apical branching begins, or None if there
+        is a problem
+    '''
+    apical = [neurite for neurite in morph.neurites
+              if nm.NeuriteType.apical_dendrite == neurite.type]
+
+    if not apical:
+        L.warning('No apical found')
+        return None
+    elif 1 < len(apical):
+        L.warning('Too many apical dendrites')
+        return None
+
+    apical = apical[0]
+
```

Comment:
The distance is computed twice for the same points.
You could keep it in a dict:

`leaf_distances = {leaf: point_dist2(leaf.points[-1, COLS.XYZ], morph.soma.center) for leaf in pical.root_node.ileaf()}
max_distance2 = max(leaf_distances.values())
min_distance2 = max_distance2 * (1 - tuft_percent / 100.) ** 2
parents = [set(leaf.iupstream) for leaf, dist in leaf_distance.items() if dist > min_distances]
common_parents = set.intersection(*parents)
`

Reply:
yeah, good point, memoization ftw.

I can't say this was written w/ performance in mind, tbh